### ISSUE TITLE
ASSERT: "RunFuture.valid()" in file Processor.cpp:39

### ISSUE THREAD

**Issue by csorbakristof** [bmeaut/grainautline#42]
After loading an image if I start to draw on it (Shift+LClick), after a while I get the followings:

QObject::connect: Cannot queue arguments of type 'QQmlChangeSet'
(Make sure 'QQmlChangeSet' is registered using qRegisterMetaType().)

And many-many times:

[Warning] "Processors are already running. Starting a new computation on the PSD while a previous one is in progress is not possible." [SlotManager.cpp:239]

If I load an image and select the Query processor and click in the image, I get the following assert fail:

```
Starting E:\Projektek\marvany\build-GrainAutLine-Desktop_Qt_5_4_0_MinGW_32bit-Debug\GrainAutLine\debug\GrainAutLine.exe...
QML debugging is enabled. Only use this in a safe environment.
QObject::connect: Cannot queue arguments of type 'QQmlChangeSet'
(Make sure 'QQmlChangeSet' is registered using qRegisterMetaType().)
[Debug] Regenerating blob index. [ShowImageProcessor.cpp:88]
[Debug] Regenerating blob index. [ShowImageProcessor.cpp:88]
[Debug] QueryProcessor::MouseClicked [QueryProcessor.cpp:28]
[Fatal] ASSERT: "RunFuture.valid()" in file Processors\Processor.cpp, line 39 [qglobal.cpp:2810]
```

**Comment by csorbakristof** [comment]
Tested on commit (Sun Jul 19 23:48:13 2015) 4bc32f7ed200fd38ce0870ed6f55efa68b23892f

**Comment by szotsaki** [comment]
For `QQmlChangeSet`-related messages, see #36.

**Comment by csorbakristof** [comment]
Hú, ez gyors volt. :)
Valóban, a connect miatt nem sikerült jó helyre asszociálnom...

**Comment by szotsaki** [comment]
Fixed in e749746d67c5bc002af44b43dda8968c9d0f8c1b.
